client/state/admin-menu/selectors/index.js

```diff
@@ -12,3 +16,8 @@ export function getAdminMenu( state, siteId ) {
 
 	return state.adminMenu[ siteId ] || null;
 }
+
+export function getIsAllDomainsView( state ) {
```

Comment:
I'm not entirely confident this selector should be located here as it isn't really relevant to this slice of state. Perhaps it should remain within the component itself?

Reply:
I vote for the component in this case

Reply:
Yeah, it's hard to say. The other functions are in `client/my-sites/email/paths.js` and `client/my-sites/domains/paths.js`, and there really isn't a natural place to put something that combines them.   I'm fine w/ the component

Reply:
Agreed. I'll revise this to move it back to the component.

Reply:
Ok I've made a custom hook `useDomainsViewStatus` so as to keep the component logic free and more easily testable.